config/initializers/deprecation_warning_subscriber.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# @note For use in conjuction with setting `Rails.application.config.active_support.deprecation = :notify`.
+#   Whenever a “deprecation.rails” notification is published, it will dispatch the event
+#   (ActiveSupport::Notifications::Event) to method #deprecation.
+class DeprecationWarningSubscriber < ActiveSupport::Subscriber
+  SLACK_ALERT_TITLE = "Deprecation Warning"
+  SLACK_ALERT_CHANNEL = "#appeals-deprecation-alerts"
+
+  attach_to :rails
+
+  def deprecation(event)
+    emit_warning_to_application_logs(event)
+    emit_warning_to_sentry(event)
+    emit_warning_to_slack_alerts_channel(event)
+  end
+
+  private
+
+  def emit_warning_to_application_logs(event)
+    Rails.logger.warn(event.payload[:message])
+  end
+
+  def emit_warning_to_sentry(event)
+    Raven.capture_message(
+      event.payload[:message],
+      level: "warning",
+      extra: {
+        message: event.payload[:message],
+        callstack: event.payload[:callstack],
+        environment: Rails.env
+      }
+    )
+  end
+
+  def emit_warning_to_slack_alerts_channel(event)
+    SlackService
+      .new(url: ENV["SLACK_DISPATCH_ALERT_URL"])
+      .send_notification(event.payload[:message], SLACK_ALERT_TITLE, SLACK_ALERT_CHANNEL)
```

Comment:
From our recent UAT testing of https://github.com/department-of-veterans-affairs/caseflow/pull/19078, we found that the alerts look like this in [DSVA `#appeals-deprecation-alerts`](https://dsva.slack.com/archives/C05L7AMMV62):


<img width="632" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/3683255/8dd9249d-8a40-44b1-83a8-871516b06262">

It's not immediately obvious which environment they are coming from.

Let's add some more detail to the Slack alert title, maybe so that it reads like

```
Deprecation Warning - efolder (uat)
```

```suggestion
    slack_alert_title = "Deprecation Warning - #{APP_NAME} (#{ENV['DEPLOY_ENV']})"
    
    SlackService
      .new(url: ENV["SLACK_DISPATCH_ALERT_URL"])
      .send_notification(event.payload[:message], slack_alert_title, SLACK_ALERT_CHANNEL)
```